src/hooks/analytics/useTrackScreen.ts

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { usePostHog } from 'posthog-react-native';
+
+/**
+ * Hook to track a screen view.
+ * @param screenName - Name of the screen to track.
+ * @param properties - Additional properties to track.
```

Comment:
Please update the documetation to inform the user that the `properties` object **must** be memoized if is not a primitive value like `string` `number` or `boolean`.
Another solution if you want to handle the momizaion in this hook is  to change the of `properties` from `any` to something like `Record<string, string | number | boolean>` so that you can use `...Object.keys(properties)` and `...Object.values(values)` to populate the depency list of the `useEffect`.